Stokhos : enable std::complex and kokkos::complex for ensemble propagation

romintomasetti opened the pull request:
## What's new in this PR ?

Stokhos is now able to perform basic mathematical operations with `std::complex` and `Kokkos::Complex`.

The main fix brought by this PR is the addition of `using std::matop` in the macros of `packages/stokhos/src/sacado/kokkos/vector/Sacado_MP_Vector_ops.hpp`. This brings a behavior that is similar to `packages/sacado/src/Sacado_Fad_Ops.hpp` for instance.

Moreover, the test `packages/stokhos/test/UnitTest/Stokhos_SacadoMPVectorUnitTest.cpp` has been adapted to ensure that `Kokkos::Complex` now works with ensembles (or the other way around :smile:).

## To do before merging

- [x] Test that the fix also works with CUDA.
- [x] Fix Stokhos to use `getLocalViewDevice` instead of `getLocalView<device>`

## Side quests

As a side quest, I also added a `Stokhos_UnitBaseTestLib` (static) that contains the `int main(...)` entrypoint. Only the tests that were strictly following the sequence `MPI - Kokkos::init - Test - Kokkos::finalize - return` now link to this *test only* library.

## Notes

This PR needs:
* https://github.com/kokkos/kokkos/pull/5009.

JacobDomagala committed:
Zoltan2: Initial changes to move creation of Zoltan2's Models from Problems to Algorithms (during partition/color method)

JacobDomagala committed:
Zoltan2: Remove the need of storing Models in Zoltan2's Problems

JacobDomagala committed:
Zoltan2: Rename WrapperForCoords to AdapterWithCoordsWrapper

mperrinel committed:
Zoltan2: fixed warnings

JacobDomagala committed:
Zoltan2: Remove no longer used code

JacobDomagala committed:
Zoltan2: Move Model creation to color() for Zoltan2_algHybrid2GL

JacobDomagala committed:
Zoltan2: Move Model creation from constructor to parition method for Zoltan2_AlgQuotient

JacobDomagala committed:
Zoltan2: Remove member Kokkos views from CoordinateModel, as they're no longer used

romintomasetti committed:
stokhos(complex support) : using std::op allowing better name lookup for custom types not supported by std

romintomasetti committed:
stokhos(complex support) : adding tests for complex numbers (both std:: and Kokkos::)

JacobDomagala committed:
Zoltan2: Add Kokkos version of functions for IdentifierModel

JacobDomagala committed:
Zoltan2: Cleanup after merging changes to Zoltan2's Sphynx changes

JacobDomagala committed:
Zoltan2: Update tests for IdentifierModel

romintomasetti commented on packages/stokhos/src/sacado/kokkos/vector/Sacado_MP_Vector_ops.hpp:
```diff
@@ -150,22 +150,26 @@ namespace Sacado {                                                      \
                                                                         \
       KOKKOS_INLINE_FUNCTION                                            \
       value_type coeff(int i) const {                                   \
+        USING_OP                                                        \
```
Aren't we going to face a similar issue as in
```
  static bool isInf(const std::complex<RealFloatType>& x) {
#ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST
    using std::isinf;
#endif
    return isinf(real(x)) || isinf(imag(x));
  }
```
?

romintomasetti committed:
stokhos : introduce a unit base test library to reduce code reduncdancy

romintomasetti commented on packages/stokhos/src/sacado/kokkos/vector/belos/Belos_TpetraAdapter_MP_Vector.hpp:
```diff
@@ -355,8 +355,8 @@ namespace Belos {
       typedef typename flat_view_type::execution_space execution_space;
-      flat_view_type flat_A_view = Atmp->template getLocalView<execution_space>();
-      flat_view_type flat_C_view = Ctmp->template getLocalView<execution_space>();
+      flat_view_type flat_A_view = const_cast<MV*>(Atmp.get())->getLocalViewDevice(Tpetra::Access::ReadWrite);
+      flat_view_type flat_C_view = const_cast<MV*>(Ctmp.get())->getLocalViewDevice(Tpetra::Access::ReadWrite);
```
A bit weird that the `Atmp` and `Ctmp` are declared `RCP<const MV>` since we want to write into their local device view...

Maybe this fix could just be `Atmp->getLocalViewDevice(Tpetra::Access::ReadWrite)` and change the type of `Atmp` to `RCP<MV>` ?

etphipp replied:
Actually, I believe C is overwritten but A is not.  So `Atmp` should be `RCP<const MV>` and `Ctmp` should be `RCP<M>`, in which case I think it should be
```
typedef typename MV::dual_view_type::t_dev const_flat_view_type;
const_flat_view_type flat_A_view = Atmp->getLocalViewDevice(Tpetra::Access::ReadOnly);
flat_view_type flat_C_view = Ctmp->getLocalViewDevice(Tpetra::Access::ReadWrite);
```

romintomasetti replied:
We introduced PR
* https://github.com/trilinos/Trilinos/pull/10710

with these changes :smile: So I'll wait it is accepted to rebase on it.

maartenarnst commented:
Is it correct to say that the mathematical operations on ensembles are currently tested only on the host?  E.g. in Stokhos_SacadoMPVectorUnitTest.cpp.

romintomasetti commented:
> Is it correct to say that the mathematical operations on ensembles are currently tested only on the host? E.g. in Stokhos_SacadoMPVectorUnitTest.cpp.

I'd say it's tested on the default execution space. See for instance `typedef ::Kokkos::DefaultExecutionSpace execution_space;`. What I don't know is if Trilinos will test it for Cuda... Maybe our tests during the Docker image build will ?

maartenarnst commented:
If it's the static fixed storage that, I think, we found works by essentially allocating an array, is the execution space really used?

romintomasetti committed:
stokhos : storage string name macro

maartenarnst commented on packages/stokhos/test/UnitTest/Stokhos_SacadoMPVectorUnitTest.cpp:
```diff
@@ -474,52 +516,73 @@ struct UnitTestSetup {
+  /* Operator atan2 not supported for complex type but supported for real type. */ \
+  BINARYFUNC_UNIT_TEST(VEC, SCALAR_T, atan2, atan2, using std::atan2;, ATan2)      \
+  /* Operators min and max don't work for complex type because UnitTestSetup hard-codes the value of 'a' as a double. */ \
+  BINARYFUNC_UNIT_TEST(VEC, SCALAR_T, max  , max  , using std::max  ;, Max)        \
+  BINARYFUNC_UNIT_TEST(VEC, SCALAR_T, min  , min  , using std::min  ;, Min)        \
```
In `Kokkos_ArithTraits.hpp`, there are `min` and `max` functions for complex types.  Could be interesting to point to those?

romintomasetti replied:
If I remember well, we discussed the use of Kokkos arithmetic traits in the past but decided not to use it because some specializations had been commented out.
Why would we use it for `max`?

maartenarnst replied:
My comment was indeed not very clear.  Not sure what the best point of view is here.  One point of view could be to decide whether there are `min` and `max` functions, indeed not necessarily those from ArithTraits, that are defined sample wise and that we want to extend to ensembles and then also test the extension.   Do calls with `min` and `max` and/or 'std::min' and `std::max` work sample wise?

Other than that, the comment just above these lines says that `min` and `max` are not tested because `a` is hardcoded as a double in the test setup.  Wouldn't there be a conversion to complex type if `a` is assigned e.g. to `cx` next.

romintomasetti replied:
> Other than that, the comment just above these lines says that min and max are not tested because a is hardcoded as a double in the test setup. Wouldn't there be a conversion to complex type if a is assigned e.g. to cx next.µ

I think my comment was wrong. `min` and `max` are simply not supported for complex types :smile:

romintomasetti replied:
I agree with @etphipp that we cannot define a generic `max`/`min` operator for complex.

maartenarnst commented on packages/stokhos/test/UnitTest/Stokhos_SacadoMPVectorUnitTest.cpp:
```diff
@@ -474,52 +516,73 @@ struct UnitTestSetup {
+#ifndef KOKKOS_ENABLE_CUDA
+  namespace Complex_std   {TEST_STATIC_STORAGE(StaticFixedStorage, StaticFixedVector,   std   ::complex<double>,    std_complex_double, 8, VECTOR_UNIT_TESTS_COMPLEX_TYPE)}
+#endif
+  namespace Complex_Kokkos{TEST_STATIC_STORAGE(StaticFixedStorage, StaticFixedVector, ::Kokkos::complex<double>, kokkos_complex_double, 8, VECTOR_UNIT_TESTS_COMPLEX_TYPE)}
```
Probably necessary to put a guard around this line to carry out these tests only if Trilinos is being compiled with complex doubles.

maartenarnst replied:
Also, it would probably be interesting to ask Eric wether we should test other types of storage as well.

maartenarnst replied:
I think it's either/or `HAVE_TPETRA_INST_COMPLEX_DOUBLE` or `HAVE_TEUCHOS_COMPLEX`.  Probably interesting to check the cmake logic to see what gets set and whether one would be better than the other.

romintomasetti replied:
Let's choose the teuchos one 😊

maartenarnst replied:
OK

etphipp replied:
Just make sure it is the same macro that defines the needed specializations.  You will probably need to use different macros for different kinds of specializations/instantiations, e.g., something for Stokhos itself (`HAVE_TEUCHOS_COMPLEX` should be fine as I think that is how Sacado does it), something else for any Kokkos specializations (should be the same on Kokkos uses), and `HAVE_TPETRA_INST_COMPLEX_DOUBLE` for Tpetra and downstream instantiations.

And you only need to due static, fixed storage.  I don't believe we even instantiate ensembles on the dynamic storage type anymore (it was just an intermediate step for building the PCE scalar type).

etphipp replied:
Yes, although I think you would also disable `std::complex` for HIP.

romintomasetti replied:
OK, let's remove `std::complex` for HIP as well :smile:

maartenarnst commented on packages/stokhos/test/UnitTest/Stokhos_SacadoMPVectorUnitTest.cpp:
```diff
@@ -474,52 +516,73 @@ struct UnitTestSetup {
+{
+  namespace Double        {TEST_STATIC_STORAGE(StaticFixedStorage, StaticFixedVector,                   double ,                double, 8, VECTOR_UNIT_TESTS_REAL_TYPE   )}
+// Skip std::complex when using CUDA, because std::complex isn't supported in that case.
+#ifndef KOKKOS_ENABLE_CUDA
+  namespace Complex_std   {TEST_STATIC_STORAGE(StaticFixedStorage, StaticFixedVector,   std   ::complex<double>,    std_complex_double, 8, VECTOR_UNIT_TESTS_COMPLEX_TYPE)}
+#endif
```
Not sure wether `KOKKOS_ENABLE_CUDA` is the guard to use.  It looks like the code runs on the host only.  So Cuda may not be an issue here.  An issue might be wether a problem arises if `nvcc` is compiling the host code.  If we don't want `nvcc` to compile this line, then we could use `__CUDACC__` as the guard?

romintomasetti replied:
If I guard with `__CUDACC__` then I don't have the `std::complex` test anymore... shouldn't I use __CUDA_ARCH__?
* https://stackoverflow.com/a/8809924

romintomasetti replied:
So, since guarding with `__CUDACC__` has the same impact as guarding with `KOKKOS_ENABLE_CUDA`, I propose to use `KOKKOS_ENABLE_CUDA` ;)

maartenarnst replied:
OK.  The sentence that you added also makes it more clear why we are disabling the line if the compilation is a compilation with cuda.

romintomasetti replied:
I think you're right, `Kokkos::complex` is always shipped as part of Kokkos. So let's indeed test it whatever the space.

maartenarnst commented on packages/stokhos/test/UnitTest/Stokhos_UnitTestHelpers.hpp:
```diff
@@ -142,10 +143,10 @@ namespace Stokhos {
     for( int i = 0; i < n; ++i ) {
-      ValueType err = std::abs(a1.coeff(i) - a2.coeff(i));
+      ValueType err = abs(a1.coeff(i) - a2.coeff(i));
       ValueType tol =
-        abs_tol + rel_tol*std::max(std::abs(a1.fastAccessCoeff(i)),
-                                   std::abs(a2.fastAccessCoeff(i)));
+        abs_tol + rel_tol*std::max(abs(a1.fastAccessCoeff(i)),
+                                   abs(a2.fastAccessCoeff(i)));
```
This is probably an important one.  We should be sure that we don't go through a conversion to int.  One option could be to instantiate Kokkos::ArithTraits for the scalar type and use its `abs` function?

romintomasetti replied:
Done.

maartenarnst replied:
OK

etphipp replied:
My guess is Kokkos doesn't inject overloads into the `std` namespace and uses `Kokkos::ArithTraits` instead, so this is probably the correct solution.

romintomasetti replied:
Ok. So we keep the solution "use Kokkos arith traits".

etphipp commented on packages/stokhos/test/UnitTest/Stokhos_UnitTestBase.cpp:
```diff
@@ -0,0 +1,20 @@
+
+    Kokkos::initialize(argc,argv);
+
+    int res = Teuchos::UnitTestRepository::runUnitTestsFromMain(argc, argv);
```
That would be good.  It is unfortunate that is not the default (I can't imagine a case where you would not want to do that).

romintomasetti replied:
Ok, I will add that ;)

etphipp commented:
> > Is it correct to say that the mathematical operations on ensembles are currently tested only on the host? E.g. in Stokhos_SacadoMPVectorUnitTest.cpp.
> 
> I'd say it's tested on the default execution space. See for instance `typedef ::Kokkos::DefaultExecutionSpace execution_space;`. What I don't know is if Trilinos will test it for Cuda... Maybe our tests during the Docker image build will ?

The tests themselves just run on the host.  They would have to be run inside some form of Kokkos parallel operation to run on the device.

etphipp commented:
> If it's the static fixed storage that, I think, we found works by essentially allocating an array, is the execution space really used?

That is right.  For static, fixed storage, the execution space doesn't matter.  This was a poor design decision on my part.

cgcgcg committed:
Xpetra ThyraUtils: Fix bad test pattern

Instead of dynamic_cast(., false) followed by is_null(), used
dynamic_cast(., true) as the error message is better.

cgcgcg committed:
Xpetra ThyraUtils: Add helper functions

cgcgcg committed:
MueLu: Allow passing more matrices as user data

cgcgcg committed:
MueLu Thyra: Convert Operators to FakeRowMatrices

This should allow to run at least some smoothers, even the RowMatrix
interface is incomplete.

cgcgcg committed:
MueLu Ifpack2Smoother Hiptmair: Allow use of operators

cgcgcg committed:
Panzer MiniEM: Simplify input deck, fully assembly geometric p-coarsen hierarchy

cgcgcg committed:
Panzer: Add evaluator GradBasisDotTensorTimesVector

cgcgcg committed:
Panzer MiniEM: Stop running Augmentation solver

cgcgcg committed:
Panzer MiniEM: Bump up CG solver its to work for higher order

cgcgcg committed:
Tpetra Operator: Add optional method for diagonal extraction

cgcgcg committed:
Ifpack2 Chebyshev: Use RowMatrix::getRowMap

JacobDomagala committed:
Zoltan2: Adapt Zoltan2_AlgMetis to take Adapter instead of Model

JacobDomagala committed:
Zoltan2: Add unittest for CoordinateModel::getCoordinatesKokkos

JacobDomagala committed:
Zoltan2: Fix IdentifierModel unittest

cgcgcg committed:
MueLu Thyra adapter: Pass diagonal if Tpetra Operator has one

cgcgcg committed:
Panzer MiniEM Utils: Write diagonal if operator has one

cgcgcg committed:
Panzer MiniEM: Add missing pragmas, remove unneeded includes

cgcgcg committed:
Panzer MiniEM: Matrix-free interpolation

cgcgcg committed:
MueLu: Move and rename OperatorSmoother -> RefMaxwellSmoother

cgcgcg committed:
Teko explicitTranspose: Return non-const Tpetra matrix

cgcgcg committed:
MueLu Stratimikos adapter: Convert Diagonal op passed as LinearOp

seanofthemillers committed:
Changes:

- Chebyshev is now able to use the native SpMV operators via an input file parameter

- Added cleanup function to Monitor timer names to remove escaped double quotes (fixes tracing issues)

cgcgcg committed:
Panzer MiniEM: Handle corner cases for matrix-free interpolations

cgcgcg committed:
Panzer STK: Add access to data as Kokkos views

cgcgcg committed:
Panzer: Fix UVM dependence in GradBasisDotVector

cgcgcg committed:
Panzer MiniEM: Add computation of L2 errors, stop using MPI_COMM_WORLD

cgcgcg committed:
Panzer MiniEM: Remove unneeded evaluators

cgcgcg committed:
Panzer: Fix issues in GradBasisDotVector and DivBasisTimesScalar evaluators

cgcgcg committed:
Panzer ModelEvaluator: Add a timer to create_W_op

It's not covered by any other timer, and can take significant time.

cgcgcg committed:
Panzer MiniEM: Add timer for auxiliary model eval

cgcgcg committed:
MueLu&Xpetra: Remove unused typedefs

cgcgcg committed:
MueLu: Rename FakeTpetraRowMatrix to TpetraOperatorAsRowMatrix

cgcgcg committed:
MueLu RefMaxwell: Refactor Stratimikos subsolves

cgcgcg committed:
MueLu Utilities: Switch RealValuedToScalarMultiVector from magnitudeType to coordinateType

cgcgcg committed:
Stokhos: ETI for additional MueLu classes

cgcgcg committed:
MueLu interface test: Replace deprecated Kokkos flag

JacobDomagala committed:
Zoltan2: Remove compile warnings

cgcgcg committed:
MueLu RAPFactory: Skip reuse of graphs on GPU

Tpetra is currently running reuse on host

cgcgcg committed:
MueLu Stratimikos example: Don't dump LHS on every solve

cgcgcg committed:
MueLu CoalesceDropFactory_kokkos: Remove old parameter and warning

cgcgcg committed:
MueLu SaPFactory_kokkos: Don't print debug info

cgcgcg committed:
MueLu: Do not print warnings for things that are not unexpected

cgcgcg committed:
MueLu: Don't get a coarse solver factory unless we want to use it

cgcgcg committed:
MueLu: Rebase gold files

cgcgcg committed:
MueLu UncoupledAggregationFactory*: Don't print redundant information

JacobDomagala committed:
Zoltan2: Check for GraphModel type in Zoltan2::createPartitioningProblem

trilinos-autotester committed:
Merge Pull Request #11122 from dridzal/Trilinos/rol-trilinos-sync

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: ROL Trilinos Sync 06-Oct-2022
PR Author: dridzal

gsjaardema committed:
SLICE Snapshot

Slice: Add line_decomp option to create lines of elements from named surface(s) and keep all elements in a line on the same processor rank.
Slice: Add option to output decomposition to map or field
Slice: Clean up debug flags

gsjaardema committed:
IOSS Partial Snapshot

* Rename DEBUG() to DebugOut(), WARNING() to WarnOut() -- reduce possible symbol conflict
* Clean out doxygen commands causing warnings.
* Use doctest testing framework instead of catch2
* Clean out blank lines, trailing whitespace
* New clang-format version changes formatting in some areas

* Better handling of optional exodus node and element maps
* Detect if parallel output being done to nfs filesystem -- warn about possible issues.
* Clean up ioss api, remove some ex_entity_type arguments.
* Add `is_shell()` to element topology routines
* Remove unuses `instance_` from element topology routines
* Add ChainGenerator functions to generate lines/columns/chains of elements for use in line solvers
* Add element permutation class for use primarily in TextMesh currently.
* Better topology matching when reading exodus files from external sources
* Add option to erase fields of a specified role from a FieldManager
* Check and warn/error if duplicate incompatible fields defined on an entity.
* For serial runs, MPI Comm returns compatible with siMPI defines
* Better field storage recognition; allowing ignoring `Real[N]` type fields
* Detect whether integer too big to store as a real without lossage.

* io_modify can scale/offset time; rename entities
* fmt: Eliminate compiler warning
* TextMesh refactor, add capability

* exodus.py - clean up, more pythonic

cgcgcg committed:
Geminga: Avoid build error in Tacho tests

cgcgcg committed:
Tpetra: Silence warning

gsjaardema committed:
IOSS: Eliminate compiler warnings

trilinos-autotester committed:
Merge Pull Request #11114 from mperego/Trilinos/intrepid_loosen_tolerance

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Intrepid2: loosen tolerance of HVol basis equivalent test
PR Author: mperego

cgcgcg committed:
Xpetra: Fixes for ETI=off

cgcgcg committed:
Stratimikos: Avoid circular include

gsjaardema committed:
IOSS: Fix compiler warning on cuda build

gsjaardema committed:
SEACAS: Fix pepu build

Fix the ordering of the `mpi.h` include so it comes prior to the `epu.C` include.

ccober6 committed:
Tempus: Minor Fixes to IntegratorBasic

 * Added feature to terminate the time integration if the
   time step size is equal to the minimum time step size AND
   the step failed.  In this situation, the integrator does
   not have any recourse, but to fail again so terminate the
   simulation.
 * Added Integrator::isInitialized().
 * Removed the tempus/test/Integrator directory and files, and
   moved the tests to the IntegratorBasic unit test.
 * Cleaned up the use of std::cout versus out in the tests and
   unit tests.
 * Cleanup some of the TRIBITS_COPY_FILES_TO_BINARY_DIR in the
   unit_test directory, so dependencies were correct.

cgcgcg committed:
Teuchos: Fix parallel XML reader to correctly handle special characters

gsjaardema committed:
IOSS: Disable parallel blob test if not pnetcdf

cgcgcg committed:
Ifpack2: Silence warning

cgcgcg committed:
Amesos2: Fix failing Epetra test

DaveBeCoding committed:
Create table_Automation.py

DaveBeCoding committed:
TODO -> CREATE SUBPROCESS TO CAPTURE OUTPUT AND PIPE/POPEN INTO <MDV>

work on todo item(s) all needed requirements have been completed (https://github.com/trilinos/Trilinos/pull/11105)

trilinos-autotester committed:
Merge Pull Request #11128 from trilinos/Trilinos/gsjaardema-patch-4

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: SEACAS: Fix pepu build
PR Author: gsjaardema

iyamazaki committed:
FROSch : use Kokkos::View instead of Techos::Array

cgcgcg committed:
Merge pull request #11131 from cgcgcg/teuchosXMLfix

Teuchos: Fix parallel XML reader to correctly handle special characters

iyamazaki committed:
FROSch : throw if MV is null

ccober6 committed:
Merge pull request #11130 from trilinos/11129-Tempus-IntegratorBasic-Termination-Improvement

Tempus: Tempus IntegratorBasic Termination Improvement

DaveBeCoding committed:
updates on master for PR11105

gsjaardema committed:
The fmt dependency should not have been removed.

cgcgcg committed:
Merge pull request #11132 from cgcgcg/amesos2epetraTest

Amesos2: Fix failing Epetra test

trilinos-autotester committed:
Merge Pull Request #11092 from boahc077/Trilinos/github_actions_token_permission

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: ci: add minimum GitHub token permissions for workflows
PR Author: boahc077

brian-kelley committed:
Tpetra TSQR: fix lwork in LAPACK tester

compute_QR and compute_explicit_Q each have their own function to
compute the work length (lwork). Since the TSQR test uses the same work
array for both computations, its length needs to be the max of the two lworks.

There is already a function lworkQueryLapackQr to do this max, but it
can't be used with cuSOLVER because cusolverDnXorgqr_bufferSize needs
the actual A/tau pointers.

Fixes #10847.

jennloe committed:
Merge pull request #11094 from jennloe/tpetraStratimikosEx

Stratimikos: New Tpetra stratimikos ex/test

brian-kelley committed:
Framework: Re-enable two TSQR tests that are now fixed

gsjaardema committed:
IOSS: Remove unneeded fmt includes; see if fixes Trilinos build

iyamazaki committed:
FROSch : fix a bug causing the segfault
 'assembledView(k+Offset_i, j) = unassembledView(k, j)' should be 'assembledView(k+Offset_i, j+itmp) = unassembledView(k, j)'

csiefer2 committed:
Ifpack2: Adding BlockTriDi example

trilinos-autotester committed:
Merge Pull Request #10184 from NexGenAnalytics/Trilinos/zoltan-refactor-of-models-and-adapters

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Zoltan2: Move creation of Models from Problems to Algorithms (during solve/color method)
PR Author: JacobDomagala

trilinos-autotester committed:
Merge Pull Request #11127 from gsjaardema/Trilinos/SEACAS-Snapshot-Oct-7-2022

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Seacas snapshot oct 7 2022
PR Author: gsjaardema

brian-kelley committed:
Merge pull request #11135 from brian-kelley/Fix10847

Tpetra TSQR: fix lwork in LAPACK tester

cgcgcg committed:
MueLu: Move Xpetra interfaces from adapters to main library

This will allow to use RefMaxwell as a coarse solver.

csiefer2 committed:
Drivers: Update to lightsaber

cgcgcg committed:
MueLu: Merge main and interface libraries

DaveBeCoding committed:
refactoring code -> PR#11105 comments

tjfulle committed:
remove errant #endif

iyamazaki committed:
FROSch : add HalfPrecisionOperator

tjfulle committed:
remove dangling #endif

iyamazaki committed:
Merge pull request #11133 from iyamazaki/frosch-fix

Frosch : fix how MV is accessed from Kokkos execution space

alanw0 committed:
STK: Snapshot 10-14-22 07:12

trilinos-autotester committed:
Merge Pull Request #11145 from trilinos/Trilinos/stk-snapshot

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: STK: Snapshot 10-14-22 07:12
PR Author: alanw0

iyamazaki committed:
Stratimikos : add achievedTol to Belos Tpetra solver

ndellingwood committed:
rol: remove volatile from join in ROL_TpetraMultiVector.hpp

Compatibility update following kokkos@3.7
See kokkos/kokkos#4077 kokkos/kokkos#5194

csiefer2 committed:
Merge pull request #11121 from trilinos/fix-11005

Tpetra: Remove HAVE_TPETRA_THROW_EFFICIENCY_WARNINGS

trilinos-autotester committed:
Merge Pull Request #11138 from trilinos/Trilinos/csiefer-773c5b8

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Ctest/Drivers: Update to lightsaber
PR Author: csiefer2

trilinos-autotester committed:
Merge Pull Request #11146 from ndellingwood/Trilinos/rol-rm-volatile-join

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: rol: remove volatile from join in ROL_TpetraMultiVector.hpp
PR Author: ndellingwood

iyamazaki committed:
FROSch : remove compiler warnings "warning: integer conversion resulted in a change of sign"

iyamazaki committed:
FROSch : remove more compiler warnings "warning: integer conversion resulted in a change of sign"

iyamazaki committed:
FROSch : remove more compiler warnings "warning: integer conversion resulted in a change of sign"

iyamazaki committed:
FROSch : HalfPrecision with reuse

iyamazaki committed:
FROSch : allow half-precision only for Tpetra + double + float

ndellingwood committed:
rol: remove volatile from join in ROL_TpetraBoundConstraint.hpp

Compatibility update following kokkos@3.7
See kokkos/kokkos#4077 kokkos/kokkos#5194

etphipp committed:
WIP on fixing Stokhos to run without UVM

etphipp committed:
Kokkos:  Allow as_view_of_rank_n() to be overloaded for "special" scalar
types

These are the same changes as Kokkos PR #5491 and will be overwritten in
the next Kokkos snapshot.  These changes are necessary so Sacado and
Stokhos can be updated now for the snapshot.

etphipp committed:
Sacado:  Add overloads of as_view_of_rank_n() for view arguments

etphipp committed:
Stokhos:  Add overloads of as_view_of_rank_n() for view arguments

trilinos-autotester committed:
Merge Pull Request #11149 from ndellingwood/Trilinos/rol-rm-volatile-join

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: rol: remove volatile from join in ROL_TpetraBoundConstraint.hpp
PR Author: ndellingwood

cgcgcg committed:
Geminga: Pass max-stracktrace to Valgrind to avoid Intrepid2 issues

trilinos-autotester committed:
Merge Pull Request #11151 from cgcgcg/Trilinos/valgrindFlag

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Geminga: Pass max-stracktrace to Valgrind to avoid Intrepid2 issues
PR Author: cgcgcg

bartlettroscoe committed:
Kokkos: Remove listing of undefined TPL deps (trilinos/Trilinos#11152)

Updated TriBITS 'master' now catches undefined TPLs listed as dependencies of
a TriBITS package.  (Amazingly, classic TriBITS simply silently ignored these
undefined TPLs.  See tribits/CHANGELOG.md entry in TriBITSPub/TriBITS#530.)

This works with older TriBITS and will be needed when TriBITS 'master' is
snapshotted into Trilinos 'develop'.

bartlettroscoe committed:
KokkosKernels: Remove listing of undefined TPL deps (trilinos/Trilinos#11152)

Updated TriBITS 'master' now catches undefined TPLs listed as dependencies of
a TriBITS package.  (Amazingly, classic TriBITS simply silently ignored these
undefined TPLs.  See tribits/CHANGELOG.md entry in TriBITSPub/TriBITS#530.)

This works with older TriBITS and will be needed when TriBITS 'master' is
snapshotted into Trilinos 'develop'.

bartlettroscoe committed:
Rythmos: Remove listing of undefined TPL deps (#11152)

Updated TriBITS 'master' now catches undefined TPLs listed as dependencies of
a TriBITS package.  (Amazingly, classic TriBITS simply silently ignored these
undefined TPLs.  See tribits/CHANGELOG.md entry in TriBITSPub/TriBITS#530.)

This works with older TriBITS and will be needed when TriBITS 'master' is
snapshotted into Trilinos 'develop'.

bartlettroscoe committed:
SEACAS: Remove listing of undefined TPL deps (trilinos/Trilinos#11152)

Updated TriBITS 'master' now catches undefined TPLs listed as dependencies of
a TriBITS package.  (Amazingly, classic TriBITS simply silently ignored these
undefined TPLs.  See tribits/CHANGELOG.md entry in TriBITSPub/TriBITS#530.)

This works with older TriBITS and will be needed when TriBITS 'master' is
snapshotted into Trilinos 'develop'.

bartlettroscoe committed:
Zoltan: Remove listing of undefined TPL deps (trilinos/Trilinos#11152)

Updated TriBITS 'master' now catches undefined TPLs listed as dependencies of
a TriBITS package.  (Amazingly, classic TriBITS simply silently ignored these
undefined TPLs.  See tribits/CHANGELOG.md entry in TriBITSPub/TriBITS#530.)

This works with older TriBITS and will be needed when TriBITS 'master' is
snapshotted into Trilinos 'develop'.

bartlettroscoe committed:
Zoltan2: Remove listing of undefined TPL deps (#11152)

Updated TriBITS 'master' now catches undefined TPLs listed as dependencies of
a TriBITS package.  (Amazingly, classic TriBITS simply silently ignored these
undefined TPLs.  See tribits/CHANGELOG.md entry in TriBITSPub/TriBITS#530.)

This works with older TriBITS and will be needed when TriBITS 'master' is
snapshotted into Trilinos 'develop'.

iyamazaki committed:
FROSch : fix build errors with Epetra

kuberry committed:
MueLu: Update to fix additional MueLu linking

- Completes the renaming of files affected by commit 4f6e0e89

trilinos-autotester committed:
Merge Pull Request #11113 from seanofthemillers/Trilinos/muelu_additional_control_parameters

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: MueLu: Adding parameters to control performance and tracing of AMG
PR Author: seanofthemillers

japlews committed:
Intrepid2: Fix some shadow variable warnings (#11142)

bartlettroscoe committed:
Add run_trilinos_pr_builds to list of gitdist repos

etphipp committed:
Remove KOKKOS_FUNCTION from one variant of as_view_of_rank_n overload

To fix calling host function on device errors.  Strange this error didn't crop up in the original Kokkos PR.

gsjaardema committed:
SEACAS: Add optional TPLs used by Seacas

Added a few optional TPLs that can be used with the SEACAS IOSS
library to provide additional capabilities.  These include:

* DataWarp -- Cray's burst buffer library
* Faodel -- a collection of data management tools that Sandia is developing to improve how datasets migrate between memory and storage resources in a distributed system.
* Cereal -- used by Faodel for serialization
* Catalyst2 -- Embedded visualization capability

trilinos-autotester committed:
Merge Pull Request #11155 from kuberry/Trilinos/muelu_remainder_eti

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: MueLu: Update to fix additional MueLu linking
PR Author: kuberry

kuberry committed:
Teko: fix return address of local temp object

etphipp committed:
Stokhos:  Fix more tests when UVM is disabled.

Mostly this is getting the Tpetra dual view semantics right by
deallocating host/device views when they are no longer needed.  Also
needed to convert the Ifpack2 RILUK test to use the scalar mean-based
preconditioner since it runs on the host.

etphipp committed:
Stokhos:  Fix Belos-Tpetra PCE adapter with multiple MPI procs

etphipp committed:
Stokhos:  Get 2-D diffusion examples working without UVM

These examples run on the host, so just allocate the KL diffusion function
on the host instead of the default execution space.

kuberry committed:
Merge pull request #11157 from kuberry/teko_fix_temp_address

Teko fix return of temporary address

trilinos-autotester committed:
Merge Pull Request #11154 from bartlettroscoe/Trilinos/11152-remove-undefined-tpl-deps

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Remove listing of undefined TPL deps (#11152)
PR Author: bartlettroscoe

jhux2 committed:
MueLu: fix simultaneous host/device access

Problem manifested under HIP in RefMaxwell.

etphipp committed:
Stokhos:  Fix usage of cuSPARSE for newer CUDA toolkits.

Call new SpMv, SpmMM functions instead of the old ones that were removed
in Cuda 11 (I think).

iyamazaki committed:
FROSch : add ctest with HalfPrecisionOperator

etphipp committed:
Stokhos:  Remove debugging statements that were accidentally committed

trilinos-autotester committed:
Merge Pull Request #11156 from gsjaardema/Trilinos/SEACAS-Add-Optional-TPL

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: SEACAS: Add optional TPLs used by Seacas
PR Author: gsjaardema

trilinos-autotester committed:
Merge Pull Request #11105 from DaveBeCoding/Trilinos.fork/tableautomation

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: adding automation script for daily table update
PR Author: DaveBeCoding

trilinos-autotester committed:
Merge Pull Request #11166 from trilinos/Trilinos/fix-miniem-host-device-crusher

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: MueLu: fix simultaneous host/device access
PR Author: jhux2

trilinos-autotester committed:
Merge Pull Request #11100 from cgcgcg/Trilinos/mueluMatrixFree

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Tpetra, Ifpack2, MueLu: Components for matrix-free multigrid
PR Author: cgcgcg

iyamazaki committed:
Merge pull request #11148 from iyamazaki/frosch-halfop

FROSch : add option to use HalfPrecisionOperator

etphipp committed:
Merge remote-tracking branch 'origin/develop' into stokhos_no_uvm_fixes

kliegeois committed:
Avoid double instantiation of CrsMatrix< SCALAR , LO , GO , NODE >::convert< SCALAR >

cgcgcg committed:
MueLu: Add missing include to install

kliegeois committed:
Add a constructor to Teuchos::RCP

kliegeois committed:
Add a unit test for the aliasing constructor of the Teuchos::RCP

kuberry committed:
Origin repo remote tracking branch: 'origin/master'
Origin repo remote repo URL: 'origin = git@github.com:SNLComputation/compadre.git'
Git describe: v1.5.4

At commit:

commit a9e29cc47cde924aa4080845143fad8d43bb71df
Author:  Paul Kuberry <pakuber@sandia.gov>
Date:    Thu Oct 20 09:31:28 2022 -0600
Summary: Add KOKKOS_THREADS_ARG as constexpr and CMake var

kliegeois committed:
Add documentation to the Aliasing constructor of Teuchos::RCP

ibaned committed:
add REQUIRED_HEADERS for TriBITS MPI TPL

this is required to
fix TriBITSPub/TriBITS#534

ibaned committed:
Netcdf_ENABLE_MODERN mode for TriBITS Netcdf TPL

when this CMake cache variable is set to a true
value, the TriBITS Netcdf TPL code will only look
for externally installed CMake config files for
CMake packages called netCDF and possibly hdf5.
Such files are installed by the latest versions
of NetCDF and HDF5.
This effectively allows loading NetCDF and its
HDF5 dependency using the most modern CMake
approach.

This is needed to resolve TriBITSPub/TriBITS#533

trilinos-autotester committed:
Merge Pull Request #11171 from cgcgcg/Trilinos/mueluOperatorAsRowMatrixFix

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: MueLu: Add missing include to install
PR Author: cgcgcg

cgcgcg committed:
MueLu nightlies: Attempt to not pull extra repos

alanw0 committed:
STK: Snapshot 10-21-22 11:03

cgcgcg committed:
Merge pull request #11177 from cgcgcg/mueluOperatorAsRowMatrixFix

MueLu nightlies: Attempt to not pull extra repos

trilinos-autotester committed:
Merge Pull Request #11150 from etphipp/Trilinos/fix_as_view_of_rank_n_for_sacado

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Overload Kokkos::Impl::as_view_of_rank_n for Sacado/Stokhos scalar types
PR Author: etphipp

cgcgcg committed:
Panzer MiniEM: Reduce size of higher order problems

jhux2 committed:
MueLu: avoid empty muelu-adapters library

If there are no source files (due to upstream dependencies),
don't generate the muelu-adapters library.

This fixes issue #11178.

bartlettroscoe committed:
Add include guards for *_def.hpp file (#10507)

When this file was converted from a *.cpp file to a *_def.hpp file when
Stratimikos::LinearSolverBuilder was templated in PR #10507, include guards
did not get added to this file.

etphipp committed:
Stokhos:  Fix unused variable warning

etphipp committed:
Stokhos:  Only use new cuSPARSE SpMv/SpMM functions on Cuda 11 or greater

While they exist on some versions of Cuda 10, it appears to me they
produce incorrect results in some cases, causing a unit-test failure.  I
verified the test does pass on the same architecture when using Cuda 11.

iyamazaki committed:
Tacho : fix issue with LDLt

bartlettroscoe committed:
Don't wipe out TPL_<tplName>_INCLUDE_DIRS if set by user (TriBITSPub/TriBITS#534)

See TriBITSPub/TriBITS#534

bartlettroscoe committed:
Revert "add REQUIRED_HEADERS for TriBITS MPI TPL" (TriBITSPub/TriBITS#534)

This reverts commit e49070d29fdd8c1d525bb22c52d7b2745271d7e2.

This causes configure failures.

iyamazaki committed:
FROSch : add an option to call Metis before ILU

iyamazaki committed:
FROSch : remove compiler warnings "unreachable code"

bartlettroscoe committed:
Make Netcdf depend on HDF5, fix TriBITS find modules (TriBITSPub/TriBITS#533)

This makes Netcdf depend on HDF5 using modern CMake targets.

ToDo: Finish this commit message.

iyamazaki committed:
Tacho : to avoid "requested too large team size."

rppawlo committed:
PIRO: fix unit test model evaluator for CUDA UVM off builds

rppawlo committed:
NOX: fix some cuda 11 compiler warnings

ibaned committed:
fix modern HDF5 import (TriBITSPub/TriBITS#533)

1. set the minimum version to the one I'm testing
   with
2. use HDF5_EXPORT_LIBRARIES since the targets
   defined are hard to predict

ibaned committed:
fix modern Netcdf import (TriBITSPub/TriBITS#533)

1. don't expect HDF5 target to be set.
   in the general case, Netcdf may not use HDF5,
   but also HDF5 has no predictable catch-all target
2. TPL_Netcdf_NOT_FOUND still has to be manually set
   to FALSE

jennloe committed:
Merge remote-tracking branch 'upstream/develop' into OrthoEx

trilinos-autotester committed:
Merge Pull Request #11172 from kliegeois/Trilinos/RCP_new_constructor

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Teuchos: Add a constructor to Teuchos::RCP
PR Author: kliegeois

jennloe committed:
Merge remote-tracking branch 'upstream/develop' into OrthoEx

jennloe committed:
Updated Orthog ex to return rank. Also creates its own map.

trilinos-autotester committed:
Merge Pull Request #11180 from trilinos/Trilinos/fix-empty-muelu-adapters-library

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: MueLu: avoid empty muelu-adapters library
PR Author: jhux2

trilinos-autotester committed:
Merge Pull Request #11185 from rppawlo/Trilinos/piro-fix-for-uvm-off

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Piro: fix tests for cuda 11 builds
PR Author: rppawlo

csiefer2 committed:
Ctest: lightsaber update

iyamazaki committed:
Tacho : cleaner way to avoid "out-of-bound access"

iyamazaki committed:
Tacho : query max vector / team sizes for factor

jennloe committed:
Delete unnecessary matrix file.

jennloe committed:
Remove unused variable.

jennloe committed:
Merge remote-tracking branch 'upstream/develop' into OrthoEx

iyamazaki committed:
Tacho : use same max vector length for factor & update

trilinos-autotester committed:
Merge Pull Request #11188 from trilinos/Trilinos/csiefer-e8c29ea

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Ctest: lightsaber update
PR Author: csiefer2

ibaned committed:
Netcdf modern is only used if HDF5 is modern

see TriBITSPub/TriBITS#533 and
also https://github.com/trilinos/Trilinos/pull/11175#discussion_r1004992219

etphipp committed:
Sacado:  Fix memory leaks in Rad when using DFad nested scalar type.

Rad does its own memory management, but doesn't know anything about it
when its nested scalar type is also dynamically allocated, which was
causing massive memory leaks when using, e.g., Rad< DFad<double> >.  The
only way to fix this without a major overhaul in Rad is to ask the user
to call a new function to zero out all of Rad's internally stored data,
which as a side-effect will deallocated DFad derivaive arrays.  Also
added this new function to the trad-dfad example.

For issue #7741.

trilinos-autotester committed:
Merge Pull Request #11165 from etphipp/Trilinos/stokhos_no_uvm_fixes

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Fix Stokhos tests/examples for CUDA w/o UVM and newer CUDA toolkits
PR Author: etphipp

trilinos-autotester committed:
Merge Pull Request #11162 from trilinos/Trilinos/stk-snapshot

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: STK: Snapshot 10-19-22 06:42
PR Author: alanw0

trilinos-autotester committed:
Merge Pull Request #11170 from kliegeois/Trilinos/Tpetra_ETI_CrsMatrix

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Tpetra: Avoid double instantiation of CrsMatrix< SCALAR , LO , GO , NODE >::convert< SCALAR >
PR Author: kliegeois

csiefer2 committed:
Ctest: lightsaber update

cgcgcg committed:
MueLu: Add AT PR script

trilinos-autotester committed:
Merge Pull Request #11193 from trilinos/Trilinos/csiefer-6aa3e85

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Ctest: lightsaber update
PR Author: csiefer2

csiefer2 committed:
Merge branch 'develop' into csiefer-ifpack2-blocktridi

iyamazaki committed:
Merge pull request #11183 from iyamazaki/tacho-ldl

Tacho : fix issue with LDLt

trilinos-autotester committed:
Merge Pull Request #11190 from etphipp/Trilinos/sacado_fix_rad_dfad_heap_leak

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Sacado:  Fix memory leaks in Rad when using DFad nested scalar type.
PR Author: etphipp

ibaned committed:
small improvements (TriBITSPub/TriBITS#533)

iyamazaki committed:
FastILU : use mirror view instead of host view

jhux2 committed:
MueLu: reenable libmuelu-adapters.a

If no adapters are enabled, add the object file "muelu-trivial-adapter"
to libmuelu-adapters.a.  In this way, TRIBITS_ADD_LIBRARY will always
be invoked and install necessary header files.

etphipp committed:
Stokhos:  Fixes for compiling on HIP

bartlettroscoe committed:
Copy from ATDM Trilinos builds driver

bartlettroscoe committed:
Have display unique GenConfig build names

This makes it easy to search in GitHub

bartlettroscoe committed:
Move to under framework so we can run in automated test suite

bartlettroscoe committed:
Run create_trilinos_github_test_failure_issue.py unit tests

trilinos-autotester committed:
Merge Pull Request #11175 from ibaned/Trilinos/tribits-netcdf-mpi

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Improve the TriBITS Netcdf and MPI TPLs
PR Author: ibaned

trilinos-autotester committed:
Merge Pull Request #11199 from trilinos/Trilinos/alway-build-muelu-adapters

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: MueLu: reenable libmuelu-adapters.a
PR Author: jhux2

trilinos-autotester committed:
Merge Pull Request #11182 from bartlettroscoe/Trilinos/10507-add-include-guards

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Add include guards for *_def.hpp file (#10507)
PR Author: bartlettroscoe

trilinos-autotester committed:
Merge Pull Request #11198 from trilinos/Trilinos/stokhos_hip_fixes

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Stokhos:  Fixes for compiling on HIP
PR Author: etphipp

iyamazaki committed:
Merge pull request #11191 from iyamazaki/frosch-ilu

Frosch : add an option to call Zoltan2 before local ILU

DaveBeCoding committed:
updating bug_report.md template instructions

csiefer2 committed:
Ifpack2: Fixing long long issue

trilinos-autotester committed:
Merge Pull Request #11203 from DaveBeCoding/Trilinos.fork/TRILINOSHD-223

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: updating bug_report.md template instructions
PR Author: DaveBeCoding

bartlettroscoe committed:
Don't unconditionally enable STK subpackages (#11205)

mayrmt committed:
MiniTensor: add doxygen page

csiefer2 committed:
Ctest: More lightsaber mods

cgcgcg committed:
Merge pull request #11194 from cgcgcg/prScript

MueLu: Add AT PR script

trilinos-autotester committed:
Merge Pull Request #11196 from trilinos/Trilinos/csiefer-ifpack2-blocktridi

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Ifpack2: Initial cut at a BlockTriDi example
PR Author: csiefer2

trilinos-autotester committed:
Merge Pull Request #11200 from bartlettroscoe/Trilinos/pr-test-failure-github-issue-gen

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Tool to automate creation of GitHub Issue for PR test failures
PR Author: bartlettroscoe

romintomasetti committed:
stokhos(test helpers): avoid using kokkos arith traits

trilinos-autotester committed:
Merge Pull Request #11208 from trilinos/Trilinos/csiefer-e255ed3

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Ctest: More lightsaber mods
PR Author: csiefer2

hkthorn committed:
Merge pull request #9878 from jennloe/OrthoEx

Belos: Add Tpetra Orthogonalization Example

kuberry committed:
Merge pull request #11174 from kuberry/compadre_v1_5_4

Compadre update to v1.5.4

cgcgcg committed:
Panzer MiniEM: Remove old comments

trilinos-autotester committed:
Merge Pull Request #11207 from bartlettroscoe/Trilinos/11205-ats2-not-always-stk

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Don't unconditionally enable STK subpackages (#11205)
PR Author: bartlettroscoe

trilinos-autotester committed:
Merge Pull Request #11206 from mayrmt/Trilinos/minitensor-add-doxygen

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: MiniTensor: add doxygen site
PR Author: mayrmt

gsjaardema committed:
SEACAS: Fix some cmake issues

trilinos-autotester committed:
Merge Pull Request #11053 from cgcgcg/Trilinos/matrixFreeMiniEM-pr

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: MiniEM: Refactor and work on matrix-free
PR Author: cgcgcg

gsjaardema committed:
Fix based on review comments

Removed `if` test altogether since `TRIBITS_ADD_EXECUTABLE_AND_TEST` takes care of same function.

trilinos-autotester committed:
Merge Pull Request #11210 from gsjaardema/Trilinos/SEACAS-Nemesis-cmake-fix

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: SEACAS: Fix some cmake issues
PR Author: gsjaardema

csiefer2 committed:
MueLu: Initial cut at matrix dump cleanup

csiefer2 committed:
MueLu: Initial cut at matrix dump cleanup

csiefer2 committed:
MueLu: More clenaup

csiefer2 committed:
Merge branch 'develop' into csiefer-ifpack2-blocktridi

GrahamBenHarper committed:
MueLu: Fix typos for matrix dumping

csiefer2 committed:
Ctest: Lightsaber updates, with help from the invincible @jhux2

trilinos-autotester committed:
Merge Pull Request #11216 from trilinos/Trilinos/csiefer-205e023

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: Ctest: Lightsaber updates
PR Author: csiefer2

trilinos-autotester committed:
Merge Pull Request #11215 from trilinos/Trilinos/csiefer-08b8b17

Automatically Merged using Trilinos Pull Request AutoTester
PR Title: MueLu: Cleanup up matrix dumping
PR Author: csiefer2

romintomasetti commented:
Closing (see https://github.com/trilinos/Trilinos/pull/10758).
